### ISSUE TITLE
SelectFilter type column filter doesn't work properly with paging

### ISSUE THREAD

**Issue by victorbr** [AllenFang/react-bootstrap-table#1305]
Have a table with paging and `SelectFilter` defined on one of the columns.
In order to reproduce:

- Navigate to the last page (or any page that would have no data after the filter)
- Apply the filter

=> Exception at line `var fieldValue = data[column.name];` of `TableBody` - `data` is undefined. When one inspects, one could see that `this.props.data` is an array having the size of one page, all elements `undefined`.
Perhaps the table should navigate to the first page before rendering?

**Comment by AllenFang** [comment]
I'll check this out. thanks

**Comment by AllenFang** [comment]
I've following simple example but still no reproduce this situation, any idea?
```js
const products = [];

const qualityType = {
  0: 'good',
  1: 'Bad',
  2: 'unknown'
};

function addProducts(quantity) {
  const startId = products.length;
  for (let i = 0; i < quantity; i++) {
    const id = startId + i;
    products.push({
      id: id,
      name: 'Item name ' + id,
      quality: 2   // all quality is unknow
    });
  }
}

addProducts(30);

function enumFormatter(cell, row, enumObject) {
  return enumObject[cell];
}

export default class SelectFilter extends React.Component {
  render() {
    return (
      <BootstrapTable data={ products } pagination>
        <TableHeaderColumn dataField='id' isKey={ true }>Product ID</TableHeaderColumn>
        <TableHeaderColumn dataField='name'>Product Name</TableHeaderColumn>
        <TableHeaderColumn dataField='quality' filterFormatted dataFormat={ enumFormatter } formatExtraData={ qualityType }
          filter={ { type: 'SelectFilter', options: qualityType } }>Product Quality</TableHeaderColumn>
      </BootstrapTable>
    );
  }
}

```

**Comment by victorbr** [comment]
Wow, it's not that simple... I think I got it but it has to do with the redux state.
What I have in addition to the simple example is a callback on `afterColumnFilter` where I fire an event. As long as there is a reducer on this event that changes *any* portion of the state (even not related to the table control at all), things break.
In the end it might be a bug on my end (perhaps I somehow cause the table to be re-rendered?). In that case I am really for having wasted your time!

**Comment by victorbr** [comment]
Yes, that's it! As a result of the event, the `render` method of the control that renders the table is called again and it seems to cause some reentrancy. Shall I close this?

**Comment by victorbr** [comment]
I have been able to solve this by using the `react-redux` `areStatesEqual` option. I supplied a function that ignores the change in the specific sub-portion of the state that is affected by the `afterColumnFilter` callback, but is not used for table rendering (naturally). As a consequence, the `render` of the table was not called as a result of filtering and the problem disappeared!
I'm closing the issue, thank you very much for your time and sorry again for this false alarm!
